pkg/controllers/management/auth/manager.go

```diff
@@ -713,17 +714,19 @@ func checkGroup(apiGroup string, rule v1.PolicyRule) bool {
 	return false
 }
 
-func (m *manager) reconcileManagementPlaneRole(namespace string, resourceToVerbs map[string]map[string]bool, rt *v3.RoleTemplate) error {
+func (m *manager) reconcileManagementPlaneRole(namespace string, resourceToVerbs map[string]map[string]string, rt *v3.RoleTemplate) error {
 	roleCli := m.mgmt.RBAC.Roles(namespace)
 	update := false
 	if role, err := m.rLister.Get(namespace, rt.Name); err == nil && role != nil {
 		newRole := role.DeepCopy()
 		for resource, newVerbs := range resourceToVerbs {
-			currentVerbs := map[string]bool{}
+			currentVerbs := map[string]string{}
 			for _, rule := range role.Rules {
 				if slice.ContainsString(rule.Resources, resource) {
 					for _, v := range rule.Verbs {
-						currentVerbs[v] = true
+						if rule.APIGroups[0] == newVerbs[v] {
```

Comment:
This is because we only want to evaluate the new verb against the single rule, right?

Reply:
Resolved by offline convo LGTM

Reply:
I had question on the same line - do we always have only 1 APIGroup entry under the rule?

Reply:
Yes because these are based off our tole templates.

Reply:
The long term goal is to refactor this to use `[]rbacv1.PolicyRule` everywhere which will get rid of a lot of this complexity such as this.